tools/mason/MasonSearch.chpl

```diff
@@ -133,16 +137,55 @@ proc masonSearch(ref args: list(string)) {
    are displayed first */
 proc rankResults(results: list(string), query: string): [] string {
   use Sort;
+  var res = rankOnScores(results);
   record Comparator { }
   proc Comparator.compare(a, b) {
     if a.toLower().startsWith(query) && !b.toLower().startsWith(query) then return -1;
     else if !a.toLower().startsWith(query) && b.toLower().startsWith(query) then return 1;
     else return 1;
   }
   var cmp : Comparator;
+  if query != ".*" then sort(res, comparator=cmp);
+  return res;
+}
+
+/* Returns a map of packages found in cache along with their scores */
+proc getPackageScores(res: list(string)) {
+  use Map;
+  const pathToReg = MASON_HOME + "/mason-registry/cache.toml";
+  var cacheExists: bool = false;
+  if isFile(pathToReg) then cacheExists = true;
+  const parse = open(pathToReg, iomode.r);
+  const cacheFile = owned.create(parseToml(parse));
+  var packageScores: map(string, int);
+  var packageName: string;
+  const defaultScore = 5;
+  var packageScore: int;
+  for r in res {
+    r = r.replace('.', '_');
```

Comment:
See printing logic above for this section. Since we're using it in 2 places, we could create a helper function:

```chpl
/* Split Pkg.0_1_0 into (Pkg, 0.1.0) */
proc splitNameVersion(package: string): (string, string) {
  var (name, version) = package.split('.');
  version = version.replace('_', '.');
  return (name, version);
}
```

Reply:
I don't think we are doing the same thing at line 164 & 102. 
At 164, we are converting `Pkg (0.1.0)` to `Pkg.0_1_0` & at 102 we are doing the opposite.

Reply:
Ah, gotcha. You could put those into helper functions or keep doing them in place then. Either way works for me.